pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,9 +120,58 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .map(m -> {
+                               Integer a = xpathExpressionArea.getText().lastIndexOf("/", m.getInsertionEnd());
+                               String s = xpathExpressionArea.getText().substring(a + 1);
+                               return Tuples.t(a, s);
+                           })
+                           .filter(t -> t._2.matches("[a-zA-Z]+"))
+                           .subscribe(e -> autoComplete(e._1, e._2));
+    }
+
+    private void autoComplete(int slashPosition, String input) {
+        System.out.println(input);
+        autoCompletePopup.getItems().clear();
+
+        XPathSuggestions xPathSuggestions = new XPathSuggestions(parent.getLanguageVersion().getLanguage());
+        List<String> suggestions = xPathSuggestions.getXPathSuggestions(input.trim());
+
+        List<MenuItem> resultToDisplay = suggestions.stream().map(m -> new MenuItem(m)).limit(5).collect(Collectors
+                                                                                                            .toList());
+
+        //TODO: Work on the implementation of the Result to be selected and added to the Code Area
+        autoCompletePopup.getItems().addAll(resultToDisplay);
+
+        if (xpathExpressionArea.getText().length() > 0) {
+
+            xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
+                if (t.isControlDown() && (t.getCode() == KeyCode.SPACE)) {
+
+                    autoCompletePopup.show(xpathExpressionArea, 500, 500);
+
+                    autoCompletePopup.setOnAction(e -> {
+                        xpathExpressionArea.getText().replace(input, "");
+                        xpathExpressionArea.replaceText(slashPosition + 1, xpathExpressionArea.getText().length() - 1, ((MenuItem) e.getTarget()).getText());
```

Comment:
You're replacing all the text after the slash position... you should only replace the input.
It seems to me, you're only testing your feature in the very restricted setting of having a simple expression like `//ClassO` that needs completion. Consider taking a more realistic query like eg AvoidStringBufferField:
```ruby
//FieldDeclaration/Type/ReferenceType/ClassOrInterfaceType[@Image = 'StringBuffer' or @Image = 'StringBuilder']
``` 
And recall that you should be able to insert a new node anywhere in the path without destroying the rest of the expression...